### DISCUSSION TITLE
issue: React hook form is crashing as soon as one form validation rule fails

### DISCUSSION THREAD

**Discussion by manuelpoelzl** [react-hook-form/react-hook-form#9705] [Q&A] [answered]
### Version Number

7.41.3

### Codesandbox/Expo snack

https://codesandbox.io/s/priceless-haze-jp5wpb?file=/src/App.js

### Steps to reproduce

The following operations are working without any problems:
- Providing correct values for both fields e.g. an username which contains an "@", and  a password; the result is a successful submit.
- Leaving both fields empty, the result is a correct validation error for both fields.

How to trigger the error:
1. Enter a string into the username field which contains an "@" and therefore is valid.
2. Do not enter a password, leave the field empty
3. Click the "Sign In" button.

This will trigger the following error: ```e.focus is not a function```.

### Expected behaviour

The form validation process should get triggered as expected, and notify the user, that the required field has no value specified.

### What browsers are you seeing the problem on?

Chrome, Edge

### Relevant log output

```shell
TypeError
e.focus is not a function
Object.focus
https://jp5wpb.csb.app/node_modules/react-hook-form/dist/index.cjs.js:1:4092
z
https://jp5wpb.csb.app/node_modules/react-hook-form/dist/index.cjs.js:1:4945
Object.ue [as _focusError]
https://jp5wpb.csb.app/node_modules/react-hook-form/dist/index.cjs.js:1:17808
eval
https://jp5wpb.csb.app/node_modules/react-hook-form/dist/index.cjs.js:1:25838
commitHookEffectListMount
https://jp5wpb.csb.app/node_modules/react-dom/cjs/react-dom.development.js:23150:26
commitPassiveMountOnFiber
https://jp5wpb.csb.app/node_modules/react-dom/cjs/react-dom.development.js:24926:13
commitPassiveMountEffects_complete
https://jp5wpb.csb.app/node_modules/react-dom/cjs/react-dom.development.js:24891:9
commitPassiveMountEffects_begin
https://jp5wpb.csb.app/node_modules/react-dom/cjs/react-dom.development.js:24878:7
commitPassiveMountEffects
https://jp5wpb.csb.app/node_modules/react-dom/cjs/react-dom.development.js:24866:3
flushPassiveEffectsImpl
https://jp5wpb.csb.app/node_modules/react-dom/cjs/react-dom.development.js:27039:3
flushPassiveEffects
https://jp5wpb.csb.app/node_modules/react-dom/cjs/react-dom.development.js:26984:14
commitRootImpl
https://jp5wpb.csb.app/node_modules/react-dom/cjs/react-dom.development.js:26935:5
commitRoot
https://jp5wpb.csb.app/node_modules/react-dom/cjs/react-dom.development.js:26682:5
performSyncWorkOnRoot
https://jp5wpb.csb.app/node_modules/react-dom/cjs/react-dom.development.js:26117:3
flushSyncCallbacks
https://jp5wpb.csb.app/node_modules/react-dom/cjs/react-dom.development.js:12042:22
eval
https://jp5wpb.csb.app/node_modules/react-dom/cjs/react-dom.development.js:25651:13
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by leapful** [comment, marked as answer]
@manuelpoelzl 

Input component (like Input, Password, ...) from PrimeReact is using prop `inputRef` to register `ref` for real DOM element of target input field. 

Just switch add `inputRef` so react-hook-form can trigger focus on the field element correctly.

Working demo: https://codesandbox.io/s/long-waterfall-8vq15c?file=/src/App.js

**Reply by manuelpoelzl** [reply]
Thanks for your response and the attached codesandbox!
Your solution works, but i was able to trigger the error again.

So if i enter a valid username and no password, and i click the button, the validation gets triggered, but if i click the button again, the same error occurs.

I´ve never usede the inputRef prop throughout my entire project, and strang enough other forms seem to work without any problems.
I prepared this codesandbox here [CodeSandbox](https://codesandbox.io/s/affectionate-lovelace-omhj7i?file=/src/App.js).

Once you click the ```Submit``` button without specifying any values, you´ll see that the ```Name``` field and the first dropdown are required.
But if i follow the same process like in the other form, one valid value and one invalid, i can not trigger this behaviour, in fact, i can´t trigger that error in this form at all.

It would be really great, if you could explain to me, why this second form here seems to be finde, although i do not use the ref or inputRef properties at all.

Thanks!

**Reply by leapful** [reply]
@manuelpoelzl 

For the first codesandbox issue, the root cause is because you're still using the whole `field` object as `{...field}` in your Input and Password component, which also registers on prop "ref" as React component instance and pollute the DOM element reference to focus of react-hook-form.

Just exclude the prop "ref" to be passed into Input and Password component will solve the problem.

Working demo: https://codesandbox.io/s/serverless-wildflower-6svw7k?file=/src/App.js

For the second codesandbox issue, you are mixing form handler of PrimeReact with react-hook-form, there for when click on button type submit with target id to the form, it's PrimeReact is responsible to perform its default validation.

Please do not use multiple form handler together, just use react-hook-form or PrimeReact to avoid confusing error.

**Reply by manuelpoelzl** [reply]
All right, now I got it!
Thanks for the explanations!
